src/_pytest/junitxml.py

```diff
@@ -207,6 +207,8 @@ def write_captured_output(self, report):
             else:
                 content = content_err
 
+            # XXX: flaky in test suite, trying to figure out why.
+            assert content, repr([content, content_log, content_err])
             if content:
```

Comment:
@nicoddemus 
Any idea why this might be Falsy during tests?
Does this happen during failures of outer tests then only?

Does it make sense to cover this branch, or should we either assert it here then always (probably a bad idea since it is covered sometimes), or just ignore it for coverage (`# pragma: no branch`)?

Reply:
Not sure, do you have an example failure?

Reply:
No.